tweak suggestion for argument-position `impl ?Sized`

lukas-code opened the pull request:
fixes this invalid suggestion:
```text
help: consider removing the `?Sized` bound to make the type parameter `Sized`
  |
1 - fn foo(_: impl ?Sized) {}
1 + fn foo(_: impl ) {}
  |
```

Commit:
tweak suggestion for argument-position `impl ?Sized`

compiler-errors reviewed (commented):
Perhaps this should instead suggest `impl Sized`? 😅

Commit:
change `std::marker::Sized` to just `Sized`

lukas-code commented:
Sure, I've updated it to suggest removing the `?` from `?Sized` (as long as there are no [macro crimes](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=89e2c7325566aee4eef9ca55faa97873) involved, in which case the suggestion is still off).

I've also added another commit to no longer print the full path of the `Sized` lang item.

compiler-errors commented:
r? @compiler-errors @bors r+ rollup

bors commented:
:pushpin: Commit 179c47f461366e92897b5c5e0b69d97de771661a has been approved by `compiler-errors`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=compiler-errors 179c47f461366e92897b5c5e0b69d97de771661a -->
<!-- homu: {"type":"Approved","sha":"179c47f461366e92897b5c5e0b69d97de771661a","approver":"compiler-errors","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:umbrella: The latest upstream changes (presumably #104455) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

lukas-code commented:
rebased

@rustbot ready

compiler-errors commented:
@bors r+

bors commented:
:pushpin: Commit b6a3f126c01d792c448f7b341352f591404bc19a has been approved by `compiler-errors`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=compiler-errors b6a3f126c01d792c448f7b341352f591404bc19a -->
<!-- homu: {"type":"Approved","sha":"b6a3f126c01d792c448f7b341352f591404bc19a","approver":"compiler-errors","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
